src/Illuminate/Database/Eloquent/UUID.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Illuminate\Database\Eloquent;
+
+use Illuminate\Support\Str;
+
+trait UUID
+{
+    /**
+     * Boot the soft deleting trait for a model.
+     *
+     * @return void
+     */
+    protected static function boot()
+    {
+        parent::boot();
+        static::creating(function ($model) {
+            if (empty($model->{$model->getKeyName()})) {
+                $model->{$model->getKeyName()} = Str::uuid()->toString();
+            }
+        });
+    }
```

Comment:
Using `boot()` will cause a trait method conflict if the consuming `Model` class wishes to define its own `boot()` behavior. `Model` supports `boot{TraitName}()` method hooks for re-usable Eloquent traits: https://github.com/laravel/framework/blob/a228123bf050167e2faef7c0bdb84bef7ed173bd/src/Illuminate/Database/Eloquent/Model.php#L252

e.g.,

```php
namespace Illuminate\Database\Eloquent;

use Illuminate\Support\Str;

trait HasUuidKey
{
    protected static function bootHasUuidKey()
    {
        // don't need to call parent::boot();

        static::creating(function (Model $model) {
            if ($model->getKey() === null) {
                $model->setAttribute($model->getKeyName(), Str::uuid()->toString());
            }
        });
    }

    // ...
}
```